### PR TITLE
[connector/datadog] Improve performance in cases with many peer tags

### PR DESCRIPTION
**Pull request by songy23**
**Description:**
Improves performance of Datadog connector when there are many peer tags and `connector.datadogconnector.NativeIngest` is enabled

**Link to tracking Issue:** 
Related to https://github.com/DataDog/datadog-agent/pull/28908

**Testing:** 
Added new benchmark tests, results:
```
% go test -bench . -benchmem  -benchtime=1000x
goos: darwin
goarch: arm64
pkg: github.com/open-telemetry/opentelemetry-collector-contrib/connector/datadogconnector
cpu: Apple M1 Max
BenchmarkPeerTags_Native-10    	    1000	  22851735 ns/op	    9147 B/op	     118 allocs/op
BenchmarkPeerTags_Legacy-10    	    1000	  27704684 ns/op	   12157 B/op	     152 allocs/op
PASS
```

### CODE REVIEW

File: .chloggen/dd-conn-peer-tags.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: datadogconnector
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Optimize Datadog connector when there are many peer tags and `connector.datadogconnector.NativeIngest` is enabled"
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: []
```
Comment (songy23) [inline .chloggen/dd-conn-peer-tags.yaml:13]:
```suggestion
issues: [34945]
```

**Comment by jackgopack4** [comment]
LGTM, makes sense based on functionality of OTLPTracesToConcentratorInputs

File: connector/datadogconnector/benchmark_test.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	for n := 0; n < b.N; n++ {
+		err = tconn.ConsumeTraces(context.Background(), genTrace())
+		assert.NoError(b, err)
+		for {
+			metrics := metricsSink.AllMetrics()
+			if len(metrics) > 0 {
+				assert.Len(b, metrics, 1)
+				break
+			}
+			time.Sleep(5 * time.Millisecond)
```
Comment (MovieStoreGuy) [inline connector/datadogconnector/benchmark_test.go:91]: Is there as reason for the sleep inside the benchmark?

**Review by dineshg13 — APPROVED** [review]

File: connector/datadogconnector/benchmark_test.go (hunk 2 above)
Comment (songy23) [reply connector/datadogconnector/benchmark_test.go]: Good catch, it doesn't have to, changed to busy-waiting

**Review by mx-psi — APPROVED** [review]

**Pull request merged**